faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -82,7 +84,54 @@ private Value() {
    * @see Codec
    */
   public final <T> Result<T> to(Codec<T> codec) {
-    return codec.apply(this);
+    return codec.decode(this);
+  }
+
+  /**
+   * Attempts to coerce this value using the {@link Decoder} class into the specified type
+   *
+   * @param clazz a class type to convert
+   * @return the {@link Result} of the coercion
+   * @see Decoder
+   */
+  public final <T> Result<T> to(Class<T> clazz) {
+    return Decoder.decode(this, Types.of(clazz));
+  }
+
+  /**
+   * Attempts to convert the specified object into the corresponding {@link Value} type
```

Comment:
"Attempts to encode an object using an {@link Encoder} into a {@link Value} type"